addons/weaponselect/XEH_preInit.sqf

```diff
@@ -16,7 +16,7 @@ private _cfgAmmo = configFile >> "CfgAmmo";
 private _cfgThrow = configFile >> "CfgWeapons" >> "Throw";
 
 {
-    private _magazines = getArray (_cfgThrow >> _x >> "magazines");
+    private _magazines = (getArray (_cfgThrow >> _x >> "magazines")) apply {_x call EFUNC(common,getConfigName)};
```

Comment:
```suggestion
    private _magazines = compatibleMagazines ["throw", _x];
```

Reply:
Doesn't work, I end up with `uns_m18white` and `uns_m18Purple` missing from `GrenadesAll`.

Reply:
I'll take a closer look tomorrow, but that's what was I gathered from examining `GrenadesAll` with and without this suggestion 🤷🏻‍♂️

Reply:
```
// abridged:
class Throw: GrenadeLauncher
	class uns_m18purpleMuzzle: ThrowMuzzle {}
	class uns_m18redMuzzle: ThrowMuzzle {}
        muzzles[] += {..."uns_m18purplemuzzle", "uns_m18redMuzzle" ... }

```
red works, purple doesn't (M in muzzle)

it's apparently case sensitive on the `muzzles[]` config?
seems like a BIS bug

Reply:
_If_ the bug is fixed, couldn't we directly use one of the following (whatever is faster)?

```sqf
GVAR(GrenadesAll) = compatibleMagazines "throw";

{
    private _ammo = getText (_cfgMagazines >> _x >> "ammo");
    private _explosive = getNumber (_cfgAmmo >> _ammo >> "explosive");
    ([GVAR(GrenadesFrag), GVAR(GrenadesNonFrag)] select (_explosive == 0)) pushBack _x;
} forEach GVAR(GrenadesAll);
```

or

```sqf
GVAR(GrenadesAll) = compatibleMagazines "throw";

GVAR(GrenadesNonFrag) = GVAR(GrenadesAll) select {getNumber (_cfgAmmo >> getText (_cfgMagazines >> _x >> "ammo") >> "explosive") == 0};
GVAR(GrenadesFrag) = GVAR(GrenadesAll) - GVAR(GrenadesNonFrag);
```

Reply:
I'm pretty sure `compatibleMagazines ["throw"]` worked fine for everything
it was just `compatibleMagazines ["throw", "x"]` that had problem

Reply:
Yeah, that was it. Works now.